dashboards/dash_key_metrics/views/lp.py

```diff
@@ -0,0 +1,103 @@
+from datetime import datetime, timedelta
+
+import streamlit as st
+
+from dashboards.utils.charts import chart_area
+
+st.markdown("# Liquidity Providers")
+
+if "chain" not in st.session_state:
+    st.session_state.chain = "All"
+if "date_range" not in st.session_state:
+    st.session_state.date_range = "30d"
+
+
+@st.cache_data
+def fetch_data(date_range, chain):
+    end_date = datetime.now()
+
+    if date_range == "30d":
+        start_date = datetime.now() - timedelta(days=30)
+    elif date_range == "90d":
+        start_date = datetime.now() - timedelta(days=90)
+    elif date_range == "1y":
+        start_date = datetime.now() - timedelta(days=365)
+    else:
+        start_date = datetime(2020, 1, 1)
+
+    core_stats_by_collateral = st.session_state.api.get_core_stats_by_collateral(
+        start_date=start_date.date(),
+        end_date=end_date.date(),
+        chain=chain,
+        resolution="28d",
+    )
+
+    return {
+        "core_stats_by_collateral": core_stats_by_collateral,
+    }
+
+
+filter_col1, filter_col2 = st.columns(2)
+
+with filter_col1:
+    st.radio(
+        "Select date range",
+        ["30d", "90d", "1y", "all"],
+        index=0,
+        key="date_range",
+    )
+with filter_col2:
+    st.radio(
+        "Select chain",
+        ["All", "Arbitrum", "Base"],
+        index=0,
+        key="chain",
+    )
+
+data = fetch_data(st.session_state.date_range, st.session_state.chain)
+
+chart_core_tvl_by_collateral = chart_area(
+    data["core_stats_by_collateral"],
+    x_col="ts",
+    y_cols="collateral_value",
+    title="TVL",
+    color="label",
+)
+chart_core_apr_by_collateral = chart_area(
+    data["core_stats_by_collateral"],
+    x_col="ts",
+    y_cols="apr",
+    title="APR",
+    color="label",
+)
+chart_core_apr_rewards_by_collateral = chart_area(
+    data["core_stats_by_collateral"],
+    x_col="ts",
+    y_cols="apr_rewards",
+    title="APR (Rewards)",
+    color="label",
+)
+chart_core_debt_by_collateral = chart_area(
```

Comment:
Now that I'm seeing this, I really wonder if this chart should exist at all... I'm not sure it's very relevant for a non-technical or general user.